packages/react-dom/src/client/ReactDOMComponent.js

```diff
@@ -422,6 +427,18 @@ export function createElement(
       // Create the script via .innerHTML so its "parser-inserted" flag is
       // set to true and it does not execute
       const div = ownerDocument.createElement('div');
+      if (__DEV__) {
+        if (enableTrustedTypesIntegration && !didWarnScriptTags) {
```

Comment:
We’ve chatted more about this. What if when trusted types are on we actually execute the script if it’s a TT object? `<script>{ttObj}</script>`. As a new feature.

Reply:
Doesn’t need to be in this PR.

Reply:
Yeah, however script would be executed only once so you would have to re-create it internally on every update.